src/util/permissions.tsx

```diff
@@ -393,3 +394,29 @@ export const getResourceOptionColumns = (type: string) => {
 
   return resourceOptionColumns[type] ?? resourceOptionColumns.default;
 };
+
+// These will need to take into consideration combined set of entitlements required for instance related actions in lxd-ui
+export const instanceEntitlementsQuery = [
+  "can_update_state",
+  // etc.
+];
+
+export const withEntitlementsQuery = (
+  isFineGrained: boolean,
+  entityType: EntityType,
+): string => {
```

Comment:
```suggestion
export const withEntitlementsQuery = (entitlements: string[]): string => {
```
I think we can simplify this by having the a `const entitlements = [...];` in the `api/instances.tsx` (and a different list in `api/profiles.tsx`, and analogue for all the entity types. Then pass the relevant list in here. We don't need a generic `EntityType` and distinguish here again. Basically pushing the splitting by types up and then this function can be pure.